app/src/main/java/com/gdgevents/gdgeventsapp/MainActivity.kt

```diff
@@ -4,44 +4,137 @@ import android.os.Bundle
+                contentColor = MaterialTheme.colors.onPrimary,
+                elevation = 8.dp,
+                modifier = Modifier.padding(WindowInsets.navigationBars.asPaddingValues()).height(56.dp)
+            ) {
+                val navBackStackEntry by navController.currentBackStackEntryAsState()
+                val currentDestination = navBackStackEntry?.destination
+                topLevelRoutes.forEach { topLevelRoute ->
+                    val isSelected = currentDestination?.hierarchy?.any {
+                        it.route == topLevelRoute.route
+                    } == true
+                    BottomNavigationItem(
+                        icon = {
+                            Icon(
+                                painter = painterResource(id = if (isSelected) topLevelRoute.selectedIcon else topLevelRoute.unselectedIcon),
+                                contentDescription = topLevelRoute.name,
+                                tint = if (isSelected) Color(0xFF1A73E8) else Color.Gray,
+                                modifier = Modifier.size(20.dp)
+
+                            )
+                        },
+                        label = {
+                            Text(
+                                text = topLevelRoute.name,
+                                style = TextStyle(
+                                    fontFamily = ProductSans,
+                                    fontWeight = FontWeight.Normal,
+                                    fontSize = 10.sp,
+                                    lineHeight = 16.98.sp,
+                                    color = if (isSelected) Color(0xFF1A73E8) else Color.Gray // Primary color for selected
+                                )
+                            )
+                        },
+                        selected = isSelected,
+                        onClick = {
+                            navController.navigate(topLevelRoute.route) {
+                                popUpTo(navController.graph.findStartDestination().id) {
+                                    saveState = true
+                                }
+                                launchSingleTop = true
+                                restoreState = true
+                            }
+                        }
                     )
                 }
             }
         }
+    ) { innerPadding ->
+        NavHost(
+            navController = navController,
+            startDestination = "home_route",
+            Modifier.padding(innerPadding)
+        ) {
+            composable(topLevelRoutes[0].route) { PlaceholderScreen(topLevelRoutes[0].name) }
+            composable(topLevelRoutes[1].route) { PlaceholderScreen(topLevelRoutes[1].name) }
+            composable(topLevelRoutes[2].route) { PlaceholderScreen(topLevelRoutes[2].name) }
+            composable(topLevelRoutes[3].route) { PlaceholderScreen(topLevelRoutes[3].name) }
+        }
```

Comment:
We need to refactor this part, we need to exactly know what is the screen name.

Reply:
Done.. replaced the placeholder with meaningful names for each screen